server/src/main/java/org/elasticsearch/common/settings/CreateKeyStoreCommand.java

```diff
@@ -21,41 +21,49 @@
 
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Arrays;
 
 import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
 import org.elasticsearch.cli.EnvironmentAwareCommand;
+import org.elasticsearch.cli.ExitCodes;
 import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
 import org.elasticsearch.env.Environment;
 
 /**
  * A subcommand for the keystore cli to create a new keystore.
  */
 class CreateKeyStoreCommand extends EnvironmentAwareCommand {
 
+    private final OptionSpec<Void> noPassOption;
+
     CreateKeyStoreCommand() {
         super("Creates a new elasticsearch keystore");
+        this.noPassOption = parser.acceptsAll(Arrays.asList("n", "nopass"), "Creates an obfuscated (not passphrase protected) keystore");
```

Comment:
Why not mimic tools like ssh-keygen which uses entering empty passphrase twice to mean no passphrase? I think this would be less confusing? If we go with an explicit command, then we should prohibit an empty passphrase to distinguish the two?

Reply:
The main driver for an option is that it is much easier to script if needed, rather then having to handle empty input twice. I also dislike hitting enter twice for no reason in existing tools (openssl, keytool, ssh-keygen) but the fact that all of them use this pattern probably adds to your argument. 

I don't think that adding a parameter means that we should prohibit emtpy passphrases on prompt. The explicit parameter has the same effect (empty passphrase) as entering an empty passphrase twice so maybe it'd be worth renaming `nopass` to `emptypass` for it to be less confusing? 

That means we do get to keep the default behavior that people are used to, but if someone needs to willingly set an empty password they can - we could throw a warning in the terminal  in this case too, to make it crystal clear.